backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ExtractOvaCommand.java

```diff
@@ -239,11 +240,11 @@ private void storeExternalData(String stdout) {
                 .collect(Collectors.toMap(part -> part[0], part -> part[1]));
         String tpmData = externalData.get("tpm");
         if (!StringUtils.isEmpty(tpmData)) {
-            vmDao.updateTpmData(getVmId(), tpmData, null);
+            vmDao.updateTpmData(getVmId(), new SecretsWrapper<String>(tpmData), null);
```

Comment:
What do you think about starting to use this object earlier, in the VM object itself, immediately after the data is fetched from the database or extracted from OVF? This will allow to find more places where this data is passed around insecurely.

Reply:
More work but a better approach.  Done.